proxy/src/proxy/mod.rs

```diff
@@ -165,7 +165,12 @@ pub async fn task_main(
                 Ok(Some(p)) => {
                     ctx.set_success();
                     let _disconnect = ctx.log_connect();
-                    match p.proxy_pass(&config.connect_to_compute).await {
+                    match tokio_metrics::TaskMonitor::instrument(
+                        &config.passthrough_task_monitor,
+                        p.proxy_pass(&config.connect_to_compute),
+                    )
+                    .await
```

Comment:
For now, could we only enable this in staging (debug logs enabled) - we currently do not know the performance impact of this instrumentation (it looks like it's almost guaranteed contention)

Reply:
Done.

Added a flag.

Reply:
If you haven't seen: https://docs.rs/tracing/latest/tracing/macro.enabled.html